store/src/directory.rs

```diff
@@ -351,15 +745,41 @@ where
     }
 
     async fn destroy_data(&self, key: &K) -> Result<(), StoreError> {
+        println!("On destroy_data (dir.rs)");
         let path = self.get_path(key);
         log::trace!("Attempting to delete data at {}", path.display());
 
-        fs::remove_file(&path).map_err(|e| {
-            StoreError::Unspecified(format!("Error removing '{}': {:?}", path.display(), e))
-        })
+        match fs::remove_file(&path) {
+            Ok(_) => (),
+            Err(e) => {
+                return Err(StoreError::Unspecified(format!(
+                    "Error removing '{}': {:?}",
+                    path.display(),
+                    e
+                )))
+            }
+        };
+        // also destroy associated metadata
+        if !self.xattr_enabled {
+            let path = set_metadata_extension_to_path(&path)?;
+            match fs::remove_file(&path) {
+                Ok(_) => Ok(()),
+                Err(e) if e.kind() == std::io::ErrorKind::NotFound => Ok(()),
+                Err(e) => {
+                    return Err(StoreError::Unspecified(format!(
+                        "Error removing associated metadata file '{}': {:?}",
+                        path.display(),
+                        e,
+                    )))
+                }
+            }
+        } else {
+            Ok(())
+        }
```

Comment:
you don't need the if/else here - removing this file and logging if there was an error (not a not found) would imply the use or xattr or not (also remove the return Err as it's the last statement anyway)
```rust
        match fs::remove_file(&set_metadata_extension_to_path(&path)?) {
            Ok(_) => Ok(()),
            Err(e) if e.kind() == std::io::ErrorKind::NotFound => Ok(()),
            Err(e) => Err(StoreError::Unspecified(format!(
                "Error removing associated metadata file '{}': {:?}",
                path.display(),
                e,
            ))),
        }
```